webapp/src/hooks/hooks.js

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {openCreateIssueModalWithoutPost} from '../actions';
+
+const createIssueCommand = '/github issue create';
+
+export default class Hooks {
+    constructor(store) {
+        this.store = store;
+    }
+
+    slashCommandWillBePostedHook = (rawMessage, contextArgs) => {
```

Comment:
What was the reasoning behind doing the slash command handling in the webapp vs doing in server side? I seems weird to me to split the command handling between server and webapp.

Reply:
I agree it's a little bit odd, this is how the implementation for Jira works too. Basically the problem here is that we want to open a modal when the comment comes in, and I was unable to see how to open the specific CreateIssue modal from the server mode. As far as I can see all the code for the modal is in the front end so unless I recreate from scratch (which would lead to having to maintain this modal in two places) then it's easiest to catch the command from the webapp and handle there.

Reply:
I flow I had in mind was "Slash command" -> "websocket event to client" -> "client opens modal" -> "create issue request to plugin". Do you think this flow is reasonable? Can it be implemented in a DRY way?

Reply:
Just looked into how this would be implemented and it turned out to be pretty simple and I think does make more sense as the server remains the only place which parses and validates the commands. I've pushed these changes so hopefully it's now looking more as you would expect.